docs/reference/Configuration.md

```diff
@@ -0,0 +1,170 @@
+# Applications Configuration
+
+Many applications require configuration parameters provided by some external
+source. Launchpad provides abstractions for dealing with the definition of the
+application parameters and their values.
+
+## Configuration parameter types
+
+Each application needs to specify which parameters are required to work.
+This specification is made by instantiating some of the configuration parameter classes:
+
+- **Mandatory**: A mandatory configuration parameter must have a value provided
+  externally. This kind of parameter is created by instantiating `MandatoryConfigurationParameter`.
+- **Optional**: An optional configuration parameter can be missing in the
+  external configuration defaulting to some specific value. This kind of
+  parameter is created by instantiating `OptionalConfigurationParameter`.
+- **Sensitive**: Sensitive parameters can be mandatory or optional (they are
+  created by decorating some of the other parameter types by sending the message
+  `asSensitive`). Its purpose is to avoid outputting the value in any log to
+  avoid leaking sensitive data.
+
+Every parameter must have a name and a summary.
+
+A parameter's value primordial type is a `String`. Optionally, a parameter can
+use a converter to parse the value and transform it
+into a proper object.
+
+Configuration parameters can also be nested in sections to group the related ones.
+
+## Configuration providers
+
+When an application starts, its configuration needs to be
+pulled from some external source. This data is retrieved using some configuration
+provider.
+
+The following providers are available:
+
+- `ConfigurationFromCommandLineProvider` tries to read the configuration parameter
+  values by parsing the command-line arguments.
+- `ConfigurationFromEnvironmentProvider` tries to read the configuration parameter
+  values from the OS environment.
+- `ConfigurationFromSettingsFileProvider` tries to read the configuration parameter
+  values by parsing the files specified using the `settings-file` option
+  of the `start` subcommand. See the [reference](CLI.md).
+
+Launchpad will evaluate the configuration providers in the following order:
+
+`Command-line -> Environment -> Settings-file`
```

Comment:
I there any kind of precedence? You can provide configuration params using more than provider?

Reply:
Yest, but I will explain that on another doc, not the reference.